streams/src/test/java/org/apache/kafka/streams/processor/internals/StateDirectoryTest.java

```diff
@@ -166,4 +166,14 @@ public void shouldListAllTaskDirectories() throws Exception {
         assertTrue(dirs.contains(taskDir2));
     }
 
+    @Test
+    public void shouldCreateDirectoriesIfParentDoesntExist() throws Exception {
+        final File tempDir = TestUtils.tempDirectory();
+        final File stateDir = new File(new File(tempDir, "foo"), "state-dir");
```

Comment:
we know it doesn't exist as `TestUtils.tempDirectory()` gives us a new empty directory.